### PR TITLE
Fix dataset_builder and add media ignores

### LINKED ISSUE #157: Tensor Preprocess device / dtype mismatch – audio on GPU bf16 vs model weights on CPU.
Tensor pre-processing on low memory hardware(?) fails due to Qwen, VAE and audio separately loaded to CPU/GPU causing dtype  mismatch in preprocess_to_tensors().

Occurs on consumer 3070 ti.

Temporary fix in, my case by forcing both models to CPU. A more correct fix would be to detect hardware/memory limit  and force both models to appropriate CUDA/CPU stack.  

around line 992 in dataset_builder.py:

 # Add batch dimension: [2, T] -> [1, 2, T]
                audio = audio.unsqueeze(0).to(device).to(vae.dtype)
                
                # Step 2: VAE encode audio to get target_latents
                with torch.no_grad():

                    # Determine the device where the VAE is currently located
                    model_device = next(vae.parameters()).device

                    # Move audio to match model device if needed
                    if audio.device != model_device:
                        audio = audio.to(model_device)

                    latent = vae.encode(audio).latent_dist.sample()
                    target_latents = latent.transpose(1, 2).to(dtype)

                # Now handle text encoding
                if 'text_input_ids' in locals() and text_input_ids is not None:

                    # Determine text encoder device
                    text_dev = next(text_encoder.parameters()).device

                    # Move ids to match text encoder device as needed
                    if text_input_ids.device != text_dev:
                        text_input_ids = text_input_ids.to(text_dev)

                    text_outputs = text_encoder(text_input_ids)
                else:
                    text_outputs = None

                latent_length = target_latents.shape[1]

note, it's taking ages to test this on my HW, sorry for late edits.

### PR DESCRIPTION
**Pull request by 1larity**
Problem:
The dataset builder failed or produced inconsistent results when processing device-related inputs due to incorrect handling of device metadata and edge cases during dataset construction.

Resolution:
Updated the dataset builder logic to correctly normalise and validate device data before use, ensuring consistent dataset generation and preventing failures during training.

fixes #157 #124

Added .gitignore entries for potentially copyright files.

### CODE REVIEW

File: acestep/training/dataset_builder.py [hunk 1]
```diff
@@ -990,10 +990,31 @@ def preprocess_to_tensors(
...
+                    text_dev = next(text_encoder.parameters()).device
+
+                    # Move ids to match text encoder device as needed
+                    if text_input_ids.device != text_dev:
+                        text_input_ids = text_input_ids.to(text_dev)
+
+                    text_outputs = text_encoder(text_input_ids)
+                else:
+                    text_outputs = None
+
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/training/dataset_builder.py]: Resolved: order of operations error (The computed text_outputs variable is assigned but never used in the visible code) and dangling text encoder/mask device alignment in later PR. #194

**Comment by Kwisss** [comment]
> Fixed in PR [#191](https://github.com/ace-step/ACE-Step-1.5/pull/191) and subsequent PR [#194](https://github.com/ace-step/ACE-Step-1.5/pull/194) bundling downstream text encoder issue, awaiting approval.

Tested your fix, and indeed the error vanished. But it seems to be stuck after that anyway? (5% cpu, vram stays empty)
Do I need to rebuild the database? I see you did 4 commits? I did:
```
git fetch origin c2d7e4c3ead7474e85df4d67252180e9a3d53a8a
git checkout c2d7e4c3ead7474e85df4d67252180e9a3d53a8a
```
but cheated with using the database I build previously.
I will retry with a subset of the database, made with your commits and report back
Ryzen 9 3900X 12C 32gb - 5060 16gb 

Thanks for your effort

EDIT:
Sad to report that with a subset and a new database it still does not take off.

**Review by ChuxiJ — APPROVED** [review]
